### PR TITLE
Uid and namespace to postgresql cluster

### PR DESCRIPTION
**Pull request by avaczi**
- Discover CRD (postgresql)
- use it in place of statefulsets (in most parts, at least)
- prepare moving PostgreSQL related bits to their own file

### CODE REVIEW

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 1]
```diff
@@ -51,6 +53,20 @@
...
+    version = "acid.zalan.do/v1"
+    endpoint = "postgresqls"
+    kind = "postgresql"
+
+
+class PostgreSQLClient(kube.Client):
+    def __init__(self, config_file_path=None, service_acc_path=kube.DEFAULT_SERVICE_ACC):
+        super().__init__(config_file_path, service_acc_path)
+
+    def get_postgresqls(self, namespace=kube.DEFAULT_NAMESPACE) -> pykube.query.Query:
```
Comment (Jan-M) [inline zmon_agent/discovery/kubernetes/cluster.py:66]: you need to do it for all namespaces

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 2]
```diff
@@ -704,6 +748,12 @@ def get_postgresql_clusters(kube_client, cluster_id, alias, environment, region,
         else:
             ss = statefulset[0]
 
+        postgresql = [pg for pg in pgs if pg['name'] == version]
```
Comment (Jan-M) [inline zmon_agent/discovery/kubernetes/cluster.py:751]: Would have been nice to convert to map here, but we have for now so few postgresqls so it is fine.

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 3]
```diff
@@ -720,15 +770,15 @@ def get_postgresql_clusters(kube_client, cluster_id, alias, environment, region,
...
             'current_replica_count': ss.get('actual_replicas', 0),
             'statefulset_error': statefulset_error,
             'deeplink1': '{}/#/status/{}'.format(hosted_zone.format('pgui', alias), version),
             'icon1': 'fa-server',
             'deeplink2': '{}/#/clusters/{}'.format(hosted_zone.format('pgview', alias), version),
             'icon2': 'fa-line-chart',
-            'uid': ss.get('metadata', {}).get('uid', ''),
-            'namespace': ss.get('metadata', {}).get('namespace', '')
+            'uid': pg.get('uid', ''),
+            'namespace': pg.get('namespace', '')
```
Comment (Jan-M) [inline zmon_agent/discovery/kubernetes/cluster.py]: this should not change imho, while true we expect it to map 1:1 but lets not assume it for the future.

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 4]
```diff
@@ -720,15 +770,15 @@ def get_postgresql_clusters(kube_client, cluster_id, alias, environment, region,
             'shards': {
                 'postgres': '{}:{}/postgres'.format(service_dns_name, POSTGRESQL_DEFAULT_PORT)
             },
-            'expected_replica_count': ss.get('replicas', 0),
```
Comment (Jan-M) [inline zmon_agent/discovery/kubernetes/cluster.py]: This should also not change, we overwrite min replica count and it should stick.

**Comment by Jan-M** [comment]
:+1:

**Comment by avaczi** [comment]
:+1:

**Pull request merged**